docs: add release procedure (#44)
lumicks/pylake: release.md

# Pylake release procedure

## Preflight check

> - Update the changelog with the date.

Reviewer: Leave a blank line after titles.

Author: 👍

- Bump the version number in `__about__.py`.

> - Check whether any dependencies have changed.

Reviewer: Add a point to review the changelog, make sure everything is clear and informative for users.

Author: Done.

## Releasing
### Pypi

> - Perform the release procedure on GH (just follow the instructions on "create a new release").

Reviewer: I think that GitHub is short enough :)

Author: Yes :D

- Include the relevant changelog entries.
- Don't forget to check "This is a pre-release".

### Conda Forge

> - Go to pypi and go to the version you just released.

Reviewer: ```suggestion
- Go to the PyPI page of the newly released version: https://pypi.org/project/lumicks.pylake/
```

Author: Changed!

- Under download files, check the sha256 hash of the tarball. You will need this for releasing on Conda Forge.

> - Use our Conda feedstock fork, set the upstream to the conda-forge master. Rebase the fork to the upstream master.
> - Create a new branch on our feedstock for your release.

Reviewer: ```suggestion
- Use our [conda-forge feedstock fork](https://github.com/lumicks/lumicks.pylake-feedstock) and create a new branch based on `upstream/master`.
```

Author: Changed!

> - Update the version in the recipe/meta.yaml if the version number changed. If the build number is not zero, set it to zero. Also update the sha256 hash.

Reviewer: ```suggestion
- Update the version in the `recipe/meta.yaml` if the version number changed. If the build number is not zero, set it to zero. Also update the sha256 hash.
```

Author: Good catch.

- If it's a re-release because some dependencies changed, increment the build number but do not change the version.
See https://conda-forge.org/docs/maintainer/updating_pkgs.html for more info.

> - Make a PR on the upstream conda forge repository. Wait for the tests to complete and pull in the changes.

Reviewer: ```suggestion
- Make a PR on the [upstream conda-forge repository](https://github.com/conda-forge/lumicks.pylake-feedstock). Wait for the tests to complete and pull in the changes.
```

Author: 👍
